apps/aecore/lib/aecore/contract/tx/contract_call_tx.ex

```diff
@@ -217,16 +216,14 @@ defmodule Aecore.Contract.Tx.ContractCallTx do
         ) :: :ok | {:error, String.t()}
   def preprocess_check(
         accounts,
-        _calls,
+        chainstate,
```

Comment:
The second argument supposed to be "Structure that holds specific transaction info in the chainstate". I am not sure if this is correct. You can check [here](https://github.com/aeternity/elixir-node/blob/master/apps/aecore/lib/aecore/tx/transaction.ex#L51).

Reply:
Yes @DanielaIvanova is right, the `@spec` of the behaviour is broken with this

Reply:
yes I know but we need the entire chainstate for this one, not only calls

Reply:
yes, but as you do it won't work with the spec

Reply:
right